test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -984,6 +984,31 @@ public void TestWriteFunctionObjectToExcel()
             Assert.AreEqual(0, list.Count);
         }
 
+        [Test]
+        public void TestOverwriteToSingleSheetExcel()
+        {
+            string openPath = Path.Combine(TestDirectory, @"core\excel\SingleSheetOverwrite.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            ViewModel.HomeSpace.Run();
+
+            var writeNode = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+
+            Assert.IsTrue(File.Exists(filename.Value));
+
+            Assert.IsTrue(writeNode.CachedValue.IsCollection);
+            var list = writeNode.CachedValue.GetElements();
+
+            Assert.AreEqual(3, list.Count);
+            AssertPreviewValue(writeNode.GUID.ToString(), new object[] { new object[] { 1 }, new object[] { 2 }, new object[] { 3 } });
```

Comment:
You might also want to verify that the xls file has only one sheet with name 'Sheet1'